src/librustc/hir/def.rs

```diff
@@ -92,9 +92,9 @@ impl DefKind {
             DefKind::Static => "static",
             DefKind::Enum => "enum",
             DefKind::Variant => "variant",
-            DefKind::Ctor(CtorOf::Variant, CtorKind::Fn) => "tuple variant",
-            DefKind::Ctor(CtorOf::Variant, CtorKind::Const) => "unit variant",
-            DefKind::Ctor(CtorOf::Variant, CtorKind::Fictive) => "struct variant",
+            DefKind::Ctor(CtorOf::Variant, CtorKind::Fn) => "enum tuple variant",
```

Comment:
This was named just variant because enum variant is a kind of "buttery butter" (does English has this idiom?) and because both "enum tuple variant" and "tuple enum variant" don't look good.
The change has the same vibe as the reverted addition of "constructor"s in https://github.com/rust-lang/rust/pull/65250, so I'd personally prefer to keep the old wording.

It would be also great to ensure that the book talks about variants when introducing enums.

Reply:
Reverted.